README.md

```diff
@@ -1,2 +1,461 @@
 # SafeDI
-Compile-time safe dependency injection in Swift
+
+[![CI Status](https://img.shields.io/github/actions/workflow/status/dfed/SafeDI/ci.yml?branch=main)](https://github.com/dfed/SafeDI/actions?query=workflow%3ACI+branch%3Amain)
+[![Swift Package Manager compatible](https://img.shields.io/badge/SPM-✓-4BC51D.svg?style=flat)](https://github.com/apple/swift-package-manager)
+[![codecov](https://codecov.io/gh/dfed/SafeDI/branch/main/graph/badge.svg)](https://codecov.io/gh/dfed/SafeDI)
+[![License](https://img.shields.io/badge/License-MIT-blue.svg)](https://spdx.org/licenses/MIT.html)
+
+Compile-time safe dependency injection for Swift projects. SafeDI is built for engineers who want the safety and simplicity of manual dependency injection without the overhead of boilerplate code.
+
+## Features
+
+✅ Compile-time safe
+
```

Comment:
Aesthetic need, I would maybe consider using regular markdown checkboxes here. If anything just to save you some space and get more content above-the-fold:

- [x] Compile-time safe
- [x] Thread safe
- [x] ...